app.py

```
@@ -0,0 +1,165 @@
+        which would slow down the user experience.
+        We use Streamlit's `st.cache_resource` to cache this loading step
+        and improve performance. Separating this logic also keeps our main code
+        cleaner and more modular.
+    """
+    # reading the file which contains the features the model was trained on
+    with open(FEATURE_VECTOR) as f:
+        data_columns = json.load(f)['data_columns']
+
+    # reading the file containing the pre-trained model
+    with open(PICKLED_MODEL, 'rb') as f:
+        model = pickle.load(f)
```

Ruff (0.13.1)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)